ocaml-lsp-server/src/document.ml

```diff
@@ -20,16 +20,31 @@ module Syntax = struct
   type t =
     | Ocaml
     | Reason
-
-  let of_language_id = function
-    | "ocaml.interface" -> Ocaml
-    | "ocaml" -> Ocaml
-    | "reason" -> Reason
-    | id -> failwith ("Unexpected language id " ^ id)
-
-  let to_language_id = function
-    | Ocaml -> "ocaml"
-    | Reason -> "reason"
+    | Ocamllex
+    | Menhir
+
+  let human_name = function
+    | Ocaml -> "OCaml"
+    | Reason -> "Reason"
+    | Ocamllex -> "OCamllex"
+    | Menhir -> "Menhir/ocamlyacc"
+
+  let all =
+    [ ("ocaml.interface", Ocaml)
+    ; ("ocaml", Ocaml)
+    ; ("reason", Reason)
+    ; ("ocaml.ocamllex", Ocamllex)
+    ; ("ocaml.menhir", Menhir)
+    ]
+
+  let of_language_id id =
+    match List.assoc all id with
+    | Some id -> id
+    | None -> Code_error.raise "invalid language id" [ ("id", String id) ]
```

Comment:
Extremely minor, but `"id"` clashes with Request ids. It's not a real clash, but it avoids a tiny confusion
```suggestion
    | None -> Code_error.raise "invalid language id" [ ("language_id", String id) ]
```

Reply:
Fixed.